src/app/home-page/widget/entry-box/entry-box.component.ts

```diff
@@ -0,0 +1,101 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { EntryType } from 'app/shared/enum/entry-type';
+import { EntryUpdateTime, User, UsersService } from 'app/shared/openapi';
+import { UserQuery } from 'app/shared/user/user.query';
+import { debounceTime, finalize, takeUntil } from 'rxjs/operators';
+import { MyWorkflowsService } from 'app/myworkflows/myworkflows.service';
+import { RegisterToolService } from 'app/container/register-tool/register-tool.service';
+import { Base } from 'app/shared/base';
+import { Dockstore } from 'app/shared/dockstore.model';
+
+@Component({
+  selector: 'app-entry-box',
+  templateUrl: './entry-box.component.html',
+  styleUrls: ['./entry-box.component.scss'],
+})
+export class EntryBoxComponent extends Base implements OnInit {
+  Dockstore = Dockstore;
+  @Input() entryType: EntryUpdateTime.EntryTypeEnum;
```

Comment:
Suggest you change this to `@Input() entryType: typeof EntryUpdateTime.EntryTypeEnum.TOOL | typeof EntryUpdateTime.EntryTypeEnum.SERVICE | typeof EntryUpdateTime.EntryTypeEnum.WORKFLOW;`, because your code below assumes it's never set to an APPTOOL.